Automattic/jetpack
# Docs: consolidate developer documentation

#### Changes proposed in this Pull Request:

Following feedback from new contributors, let's try to consolidate our docs a bit, to highlight the CLI tool and its setup, as well as clearly separate the Docker docs from the main development set up docs.


#### Jetpack product discussion

* p1625740352327600-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Check for typos
* Check that the links are correct.
* Check if the changes make sense from a new contributor's perspective
